### ISSUE TITLE
Chat channels listing is not ordered by most recent activity

### ISSUE THREAD

**Issue by peppy** [ppy/osu-web#11679]
It should be, on initial load and when new messages come in.

We already have `last_message_id` in `osu_chat.channels`, so it may be quite feasible to make this happen without too much stress.

Of note, it should also be returned in this order by the API if possible.

### Discussed in https://github.com/ppy/osu/discussions/18461

<div type='discussions-op-text'>

<sup>Originally posted by **respektive** May 28, 2022</sup>
On osu-web the direct messages are order by recent activity (newer DMs are at the top), while in lazer it's seemingly random.

Direct messages order on web:
![osu-web](https://pek.li/u385uq.png)
vs. order in lazer:
![lazer](https://pek.li/rm4gck.png)

This isn't really an issue when you only have a few DMs open like I do in the screenshots, but the more chats you open the harder it gets to keep track of which chats are actually relevant.</div>

**Comment by nanaya** [comment]
channels are currently sorted differently by type (last message id for pm and alphabetically for the rest) so osu-web just sort them on the frontend when doing the grouping.

**Comment by peppy** [comment]
So is the correct resolution here for lazer to do it client side?

**Comment by nanaya** [comment]
yeah, unless we're changing the order to be by last message id for all channel types.

even then it seems still easier to have sorting on frontend as new message comes in. it'd save the need to modify the channel array without it ending up just being a full on sort.

**Comment by peppy** [comment]
Okay thanks, moving back to other repo.
